### ISSUE TITLE
property:synonyms

### ISSUE THREAD

**Issue by nielsklazenga** [tdwg/tcs2#65]
### synonyms (property)

|    |    |
|----|----|
| Label | Synonyms |
| Definition | Names considered to apply to the same taxon as the accepted name (`taxonName`) |
| Usage notes |  |
| Comments |  |
| Required | No |
| Repeatable | No |
| Constraints | Object |

#### Mapping

|    |    |
|----|----|
| TCS | – |
| TDWG Ontology | – |

**Comment by nielsklazenga** [comment]
In my world view synonyms are most comparable with `taxonName`. On the other hand, when using TNUs, `synonyms` could be seen as the inverse property of `acceptedNameUsage`.

**Comment by nielsklazenga** [comment]
The documentation of the `has synonym` relationship type says the following:

> The target concept is considered a synonym of the current concept. This is an ambiguous relationship. It can mean: 1) a nomenclatural relationship where all that is implied is that the type of the target concept is included in the current circumscription. This is more precisely expressed as a SpecimenCircumscription (for heterotypic synonyms) or as TaxonName basionym relationships (for homotypic synonyms) 2) a concept relationship where some part of (or all of) the target concept is included in the current circumscription. This is more precisely expressed using the set relationships such as 'is congruent to'. This is intended for handling legacy data.

I think 'synonym' can only be used in the first sense, but agree that it is often used in the second sense. In any case, we have agreed that this should be split up.

I agree that synonyms are better expressed as (part of) a `specimenCircumscription` (rather than as Taxon Relationships), but, if that goes for synonyms, it goes for accepted names as well. Also, the type specimens are not always given. Therefore, it makes the most sense to have a property `synonym` on the Taxon Concept, besides `taxonName` (which is the accepted name).

For the second sense, we have added `intersects` as a relationship type, for situations where one cannot confidently tell whether the relationship type is `congruent`, `includes`, `includedIn` or `overlaps`.

**Comment by Archilegt** [comment]
Definition: Name considered to apply to the same taxon as the `acceptedNameUsage`.
  
See `acceptedNameUsage` (http://rs.tdwg.org/dwc/terms/acceptedNameUsage), formerly `acceptedScientificName`, now deprecated.

Synonyms are neither properties of specimens nor of taxonomic concepts, they are properties of scientific names and should be expressed as relations between scientific names. Also, `synonym` alone cannot be seen as the inverse property of `acceptedNameUsage`, because all accepted names with synonyms are also synonyms. We need a controlled vocabulary for relationships among synonyms, e.g., `valid synonym of`,  `invalid synonym of`, `junior synonym of`...

**Comment by nielsklazenga** [comment]
The definition is actually taken verbatim from the glossary in the Vienna Code (2006) of International Code of Botanical Nomenclature. I do not think that definition will have changed in later editions of the Code, as there is nothing wrong with it.

'synonym' is the "inverse" of 'accepted name' (our `taxonName` (#2)), not `dwc:acceptedNameUsage`. Both are properties of Taxon Concept, as while it is nomenclatural business rules that decide which name is the accepted name and which the synonym(s), to know whether or not two names apply to the same taxon in the first place requires a taxonomic opinion, ergo a Taxon Concept. And yes, the accepted name is a synonym in the dictionary sense, which is why `taxonName` and `synonym` should be dealt with in the same way, i.e. as properties of Taxon Concepts.

`dwc:acceptedNameUsage` has caused many problems in the past, so we will do well to avoid it. The biggest problem with it is that it is used as the disposition of a name by a third party (i.e. the author of the data set) of a name that is by another author (the author of the taxon concept). Whether this is by design, or because it is not made clear that it should not be used that way, I do not know, but we want to be clear in TCS. 

Also, `dwc:acceptedNameUsage` is a string. We are leaving it up to implementations (as we are not going to agree) whether the object of `tcs:synonym` is a Taxon Name or a Taxon Concept, but it is definitely an object, not a string.

**Comment by nielsklazenga** [comment]
> Also, `dwc:acceptedNameUsage` is a string. We are leaving it up to implementations (as we are not going to agree) whether the object of `tcs:synonym` is a Taxon Name or a Taxon Concept, but it is definitely an object, not a string.

Obviously, we are not leaving this up to applications anymore. `tcs:synonym`, like `tcs:taxonName` is a TaxonName (or other type of label object).

**Comment by baskaufs** [comment]
> 'synonym' is the "inverse" of 'accepted name' (our `taxonName` ([#2](https://github.com/tdwg/tcs2/issues/2))), not `dwc:acceptedNameUsage`. Both are properties of Taxon Concept, as while it is nomenclatural business rules that decide which name is the accepted name and which the synonym(s), to know whether or not two names apply to the same taxon in the first place requires a taxonomic opinion, ergo a Taxon Concept. And yes, the accepted name is a synonym in the dictionary sense, which is why `taxonName` and `synonym` should be dealt with in the same way, i.e. as properties of Taxon Concepts.

I guess I understand the argument that whether or not two names apply to the same concept requires a taxonomic opinion. But I guess what I am wondering is if the nomenclatural business rules had decided that an accepted name had a synonym, would any concept that had that accepted name as a label automatically have all of the nomenclaturally declared synonyms as values for its synonym property? If so, that would seem to support the argument that synonym is a property of the name rather than the concept.

**Comment by nielsklazenga** [comment]
@baskaufs, there are two types of synonyms, 'homotypic' (or 'nomenclatural' or 'objective' or 'obligatory') synonyms and 'heterotypic' (or 'taxonomic' or 'subjective' or 'facultative') synonyms. Homotypic synonyms would be attributes of the TaxonName. We do not use the term `homotypic synonym` in TCS, as it would have a N! problem, but `basionym` and `replacedName` are homotypic synonyms. The term 'homotypic synonym' also comes with some extra what you called 'semantic clutter' earlier, as homotypic synonyms have to be of the same taxonomic rank. `basionym` and `replacedName` do not have this problem.

Heterotypic synonyms have different nomenclatural types, so they are only synonyms in the context of a TaxonConcept. So, as to:

> I am wondering is if the nomenclatural business rules had decided that an accepted name had a synonym, would any concept that had that accepted name as a label automatically have all of the nomenclaturally declared synonyms as values for its synonym property?

...that goes for homotypic synonyms, but not for heterotypic synonyms, so also not for every potential object of the `synonym` property.

Therefore, heterotypic synonyms are attributes of TaxonConcepts and the `synonym` property, which can be homotypic or heterotypic synonyms, has to be a property of TaxonConcept, as you could in principle do homotypic synonyms on both TaxonName and TaxonConcept, but heterotypic synonyms only on TaxonConcept.

In TCS, accepted names do not have synonyms, but TaxonConcepts have an accepted name (`taxonName`) and zero or more synonyms. This translates very well to (or from) the Darwin Core Taxon, as one could use `taxonConceptID` instead of `acceptedNameUsageID` and then use `taxonomicStatus` to indicate whether a taxonomic name usage is `accepted` or a `synonym` (or a `homotypicSynonym` or `heteroTypicSynonym`).

**Comment by nielsklazenga** [comment]
Okay, I still had an old definition in the Botanical Code in my head.

Vienna Code (2006)
> **homotypic synonym (nomenclatural synonym)**. A synonym based on the same type of that of another name in the same rank (Art. 14.4); termed an "objective synonym" in the _International Code of Zoological Nomenclature_ and the _Bacterial Code_.

Shenzhen Code (2018)
> **homotypic synonym (nomenclatural synonym)**. A name based on the same type as that of another name ([Art. 14.4](https://www.iapt-taxon.org/nomen/pages/main/art_14.html#Art14.4)); indicated by the symbol “≡” in the Appendices of the Code; termed an “objective synonym” in the _International Code of Zoological Nomenclature_.

I have always thought the rule that homotypic synonyms have to be of the same rank a bit silly and now it is gone. So the extra semantic clutter I mentioned yesterday is not there anymore. I have updated the notes of the `synonym` term.

**Comment by ghwhitbread** [comment]
But, from ICZN online(https://code.iczn.org/glossary/)
>     synonym, n.
> Each of two or more names of the **same rank** used to denote the same taxonomic taxon.
> 

And I believe all Codes require synonyms to be acceptably published (valid/legitimate/available).
